### PR TITLE
Updated `kf5` packages to version 5.106.0.

### PR DESCRIPTION
**Pull request by PawelWMS**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Recent update to Qt5 (#5491) changed the packages' APIs and caused `kf5-kcoreaddons` fail to build. Updating the whole set of `kf5` packages to the latest 5.106.0 version to remedy the issue.

PR removes the default and empty `kf5` package. No other packages have a dependency on `kf5` and the package itself declared only two explicit run-time dependencies on `kf5-filesystem` and `kf5-rpm-macros`. Removal of the package should not cause regressions even if external customers took a dependency on `kf5`, as the older version of that package remain available and will pull in the latest `kf5-filesystem` and `kf5-rpm-macros`.

###### Change Log  

- Updated `kf5*` packages to version 5.106.0.
- Updated the `kpmcore` package to work with the new versions of `kf5*` packages.
- Added a new `polkit-qt5-1` package to build the updated `kpmcore` package.
- Updated licensing information.
- Removed the default `kf5` package.

###### Does this affect the toolchain?  

No.

###### Associated issues  

- #5491

###### Test Methodology

- [Full build](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=364285&view=results).

### CODE REVIEW

**Review by mfrw — APPROVED** [review]
LGTM :)
Although, you have done a build of 10 packages, should we do a full build just to be doubly sure we do not break anything.

**Comment by PawelWMS** [comment]
> LGTM :) Although, you have done a build of 10 packages, should we do a full build just to be doubly sure we do not break anything.

We've got time, sure.

https://dev.azure.com/mariner-org/mariner/_build/results?buildId=364285&view=results

**Comment by PawelWMS** [comment]
> > LGTM :) Although, you have done a build of 10 packages, should we do a full build just to be doubly sure we do not break anything.
> 
> We've got time, sure.
> 
> https://dev.azure.com/mariner-org/mariner/_build/results?buildId=364285&view=results

@mfrw: that was a good call. Turned out `kf5.spec` needed to be updated as well despite no significant changes.

**Review by oliviacrain — APPROVED** [review]

**Comment by mfrw** [comment]
> > > LGTM :) Although, you have done a build of 10 packages, should we do a full build just to be doubly sure we do not break anything.
> > 
> > 
> > We've got time, sure.
> > https://dev.azure.com/mariner-org/mariner/_build/results?buildId=364285&view=results
> 
> @mfrw: that was a good call. Turned out `kf5.spec` needed to be updated as well despite no significant changes.

The reason for this paranoia is; `learning from first hand experience`; i got bit by a seemingly un-intersting channge that broke something that i would not have remotely guessed.

**Review by mfrw — APPROVED** [review]

File: SPECS/calamares/calamares.spec [hunk 1]
```diff
@@ -231,6 +235,12 @@ install -p -m 644 %{SOURCE53} %{buildroot}%{_sysconfdir}/calamares/mariner-eula
 %{_libdir}/cmake/Calamares/
 
 %changelog
+* Tue May 23 2023 Pawel Winogrodzki <pawelwi@microsoft.com> - 3.2.11-42
+- Updating BRs after fixing "kf5" and "Qt5" builds.
+
+* Fri May 19 2023 Pawel Winogrodzki <pawelwi@microsoft.com> - 3.2.11-41
```
Comment (anphel31) [inline SPECS/calamares/calamares.spec:241]: I think you could merge these two changelog entries together, since they're part of the same PR

**Review by anphel31 — APPROVED** [review]

**Review by trungams — APPROVED** [review]

**Comment by PawelWMS** [comment]
**Note**: not merging yet - still have to verify, if the ISO installer works correctly with these changes.

**Pull request merged**